ws-client/src/client.rs

```diff
@@ -656,19 +662,22 @@ async fn background_task(
 						serde_json::from_slice::<serde_json::Value>(&raw)
 					);
 					let _ = front_error.send(Error::Custom("Unparsable response".into()));
-					return;
+					break;
 				}
 			}
 			Either::Right((Some(Err(e)), _)) => {
 				log::error!("Error: {:?} terminating client", e);
 				let _ = front_error.send(Error::Transport(e.into()));
-				return;
+				break;
 			}
 			Either::Right((None, _)) => {
 				log::error!("[backend]: WebSocket receiver dropped; terminate client");
 				let _ = front_error.send(Error::Custom("WebSocket receiver dropped".into()));
-				return;
+				break;
 			}
 		}
 	}
+
+	// Send WebSocket close reason to the server (this might fail if the server already terminated the connection).
```

Comment:
The comment seems to indicate we actually send a close reason, which it's not, really (it's always hard coded to `1000` I think?).

Reply:
yepp, bad comment I will re-phrase.